### PR TITLE
Task 149 - Navigation from unlogged dashboard to MainActivity

### PR DESCRIPTION
**Pull request by Kuba-Nalepa**
Moreover I did little fixes in login/registration/forgot password layouts like correcting them properly using custom views, changing overlapped text views, setting (working) navigation buttons

### CODE REVIEW

File: presentation/src/main/java/com/safetyheads/akademiaandroida/presentation/ui/activities/DashboardActivity.kt [hunk 1]
```diff
@@ -1,12 +1,14 @@
 package com.safetyheads.akademiaandroida.presentation.ui.activities
 
-import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
+import android.util.Log
```
Comment (Zalezny) [inline presentation/src/main/java/com/safetyheads/akademiaandroida/presentation/ui/activities/DashboardActivity.kt:4]: unused import

File: presentation/src/main/java/com/safetyheads/akademiaandroida/presentation/ui/activities/DashboardActivity.kt [hunk 2]
```diff
@@ -1,12 +1,14 @@
...
+import androidx.appcompat.app.AppCompatActivity
 import androidx.navigation.NavController
 import androidx.navigation.fragment.NavHostFragment
 import androidx.navigation.ui.setupWithNavController
 import com.safetyheads.akademiaandroida.presentation.R
 import com.safetyheads.akademiaandroida.presentation.databinding.ActivityDashboardBinding
+import com.safetyheads.akademiaandroida.presentation.ui.customviews.DashboardFragment
```
Comment (Zalezny) [inline presentation/src/main/java/com/safetyheads/akademiaandroida/presentation/ui/activities/DashboardActivity.kt:11]: unused import

File: presentation/src/main/java/com/safetyheads/akademiaandroida/presentation/ui/components/BackArrowView.kt [hunk 3]
```diff
@@ -28,10 +28,17 @@ class BackArrowView : FrameLayout {
     }
 
     private fun initView() {
-        binding = ViewArrowBinding.inflate(LayoutInflater.from(context), this, true)
+        binding = ViewArrowBinding.inflate(LayoutInflater.from(context), this)
+//        binding.imageButton.setOnClickListener()
```
Comment (Zalezny) [inline presentation/src/main/java/com/safetyheads/akademiaandroida/presentation/ui/components/BackArrowView.kt:32]: unused comment

File: presentation/src/main/java/com/safetyheads/akademiaandroida/presentation/ui/customviews/DashboardFragment.kt [hunk 4]
```diff
@@ -18,4 +22,26 @@ class DashboardFragment : Fragment() {
...
+            if(isUserLogged) {
+                binding.backButton.visibility = View.GONE
+            } else {
+                binding.backButton.customButtonListener {
+                    val intent = Intent()
+                    intent.setClassName(
+                        requireContext(),
+                        "com.safetyheads.akademiaandroida.presentation.ui.MainActivity"
+                    )
+                    startActivity(intent)
```
Comment (Zalezny) [inline presentation/src/main/java/com/safetyheads/akademiaandroida/presentation/ui/customviews/DashboardFragment.kt:42]: Please add "finish()" these activity. I think that "finish()" in MainActivity (when open DashboardActivity) will fix the error when we come from Sign in and return next

**Review by Zalezny — CHANGES_REQUESTED** [review]
Big plus for adding the back arrow component and cleaning up in the code app. As for the task itself, everything seems ok, just a few code fixes needed

**Review by Zalezny — DISMISSED** [review]

File: presentation/src/main/java/com/safetyheads/akademiaandroida/presentation/ui/signup/SignUpFragment.kt [hunk 5]
```diff
@@ -87,5 +80,25 @@ class SignUpFragment : Fragment() {
...
+                binding.eTextPassword.text.toString(),
+            )
+        }
+
+        binding.buttonBack.customButtonListener {
+            findNavController().navigateUp()
+        }
+
+        binding.signInNow.setOnClickListener {
+            findNavController().navigateUp()
```
Comment (przemek19980102) [inline presentation/src/main/java/com/safetyheads/akademiaandroida/presentation/ui/signup/SignUpFragment.kt:100]: are you sure we should do the same action as for back button?

File: presentation/src/main/res/layout/fragment_faq.xml [hunk 6]
```diff
@@ -1,138 +1,132 @@
...
             android:padding="@dimen/size_2x"
-            android:text="@string/ask_question"
-            android:textSize="11sp"
-            app:layout_constraintBottom_toBottomOf="parent"
-            app:layout_constraintEnd_toEndOf="parent"
-            app:layout_constraintStart_toEndOf="@+id/textView4" />
+            app:layout_constraintTop_toBottomOf="@id/linear_layout_faq">
+
+            <TextView
+                android:id="@+id/textView4"
```
Comment (przemek19980102) [inline presentation/src/main/res/layout/fragment_faq.xml:102]: this is not really a good name for an id

File: presentation/src/main/res/layout/fragment_faq.xml [hunk 7]
```diff
@@ -1,138 +1,132 @@
...
+            android:layout_height="48dp"
+            android:layout_marginTop="@dimen/size_3x"
+            android:layout_marginEnd="@dimen/size_3x"
+            android:src="@drawable/placeholder_avatar"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintTop_toTopOf="parent"
+            app:shapeAppearanceOverlay="@style/circleImageView" />
+
+        <com.safetyheads.akademiaandroida.presentation.ui.components.H2View
+            android:id="@+id/h2View3"
```
Comment (przemek19980102) [inline presentation/src/main/res/layout/fragment_faq.xml:32]: this is not really a good name for an id

File: presentation/src/main/res/layout/fragment_forgot_password.xml [hunk 8]
```diff
@@ -5,127 +5,132 @@
...
-        android:layout_width="wrap_content"
-        android:layout_height="wrap_content"
-        android:layout_marginTop="@dimen/size_3x"
-        android:text="@string/e_mail"
-        app:layout_constraintStart_toStartOf="parent"
-        app:layout_constraintTop_toBottomOf="@+id/tv_forgot_pass_text" />
+        <ImageView
+            android:id="@+id/logo_safetyheads"
+            style="@style/Logo_safetyheads"
+            android:layout_width="128dp"
```
Comment (przemek19980102) [inline presentation/src/main/res/layout/fragment_forgot_password.xml:28]: can't we make those 0dp and somehow align it in the layout so it looks good and it is not hardcoded?

**Review by przemek19980102 — CHANGES_REQUESTED** [review]

**Review by przemek19980102 — APPROVED** [review]

**Pull request merged**